thrifty-schema/src/main/java/com/microsoft/thrifty/schema/UserType.java

```diff
@@ -0,0 +1,56 @@
+package com.microsoft.thrifty.schema;
+
+import com.google.common.collect.ImmutableMap;
+
+public abstract class UserType extends NewThriftType implements UserElement {
+    private final Program program;
+    private final UserElementMixin mixin;
+
+    UserType(Program program, UserElementMixin mixin) {
+        super(mixin.name());
+        this.program = program;
+        this.mixin = mixin;
+    }
+
+    public String getNamespace(NamespaceScope namespace) {
+        String ns = program.namespaces().get(namespace);
+        if (ns == null && namespace != NamespaceScope.ALL) {
+            ns = program.namespaces().get(NamespaceScope.ALL);
+        }
+
+        if (ns == null) {
```

Comment:
This feels a little awkward. Most cases for us, we actually just error if there's no namespace. Just returning `null` would be nice in that it would fail faster or in a more obvious way anyway